File: cpp/runfiles_helper/RunfilesHelper.cpp
#include "cpp/runfiles_helper/RunfilesHelper.h"
#include "tools/cpp/runfiles/runfiles.h"

#include <memory>
#include <stdexcept>
#include <string>

namespace helper {
std::string getResourcePath(const std::string &resourcePath) {
  std::string error;
  using bzl_Runfiles = bazel::tools::cpp::runfiles::Runfiles;
  std::unique_ptr<bzl_Runfiles> runfiles(bzl_Runfiles::CreateForTest(&error));

  if (runfiles == nullptr) {
    throw std::runtime_error("Failed to create Runfile Instance: " + error);
  }

  return runfiles->Rlocation(resourcePath);
}

} // namespace helper

File: python/py_module/runfiles_helper.py
from rules_python.python.runfiles import runfiles

class RunfilesHelper:
    """A simple example class"""
    i = 12345

    def f(self):
        return 31
    
    def readFile(runfiles_key):
        r = runfiles.Create()
        return r.Rlocation(f"{runfiles_key}")
